html-css/document-object-model.md

```diff
@@ -110,14 +110,46 @@ Of course, there are several other ways in which you can find nodes on the page,
 
 ![Get options](/images/dom/get-option.png)
 
+#### **querySelector** and **querySelectorAll**
+```querySelector``` is an alternative to ```getElementBy``` and is the do-it-all method when it comes to element selection.  It can be used in place of any of the ```getElementBy``` methods.  The querySelector method allows us to use the CSS selector syntax that we already know and love to select elements.  ```querySelector``` will return the **first** match only and ```querySelectorAll``` will return **all** matches.
+
+Basic syntax is: ```document.querySelector("< CSS selector >")  ```
+
+
+**Select by class**
+```
+document.querySelector(".className")
+```
+**Select by id**
+```
+document.querySelector("#idName")
+```
+**Select by tag**
+```
+document.querySelector("tagName")
+```
+
+These are the most basic examples of ```querySelector```.  Because the arguments it takes are literally just CSS selectors, this enables us to be very specific with selections.  For instance, the following statement will select only ```anchor``` tags where ```src``` attribute is ```"myimage.jpg"```
+```
+document.querySelectorAll("a[src='myimage.jpg']")
+```
+This statement will select the third ```li``` in the second ```ul```:
+```
+document.querySelector("ul:nth-of-type(2) li:nth-of-type(3)")
+```
+
+
+
 ##### 3. Manipulating the DOM elements
 
-While we can certainly inspect the DOM elements and its structure, we can also change it. Below we are changing the text in the ```<h1>``` from "This is my homepage" to "This is David's homepage" using ```element.innerText```.
+While we can certainly inspect the DOM elements and its structure, we can also change it. Below we are changing the text in the ```<h1>``` from "This is my homepage" to "This is David's homepage" using ```element.textContent```.
 
 ![Before node manipulation](/images/dom/manipulate-before.png)
 
 ![After node manipulation](/images/dom/manipulate-after.png)
 
+*Please note: the refernce to* ```a.innerText``` *in this image should be* ```a.textContent```
```

Comment:
Small typo here. It would also be useful to briefly explain what the difference is and why one is preferred.